### PR TITLE
More importer work

### PR DESCRIPTION
**Pull request by nullstyle**
This PR should complete the effect importing system, with a few exceptions that are blocked by stellar-core.  

Changes to importer:
- "trade" effects are now imported
- "manage_offer" operation and "create_passive_offer" operations now import order book details
- "account_flags_updated" effect now has details
- signer_* effects are now imported
- trustline_* effects are now imported

In addition, this PR:
- Updates the bundles gems to latest versions
- Updates the fixture data to reflect the latest stellar-core
- Code cleanup
- Better error messages when parsing effect names

Known issues:
- there is no differentiation between signer_created/updated and trustline_created/updated as explained some in https://github.com/stellar/stellar-core/issues/712

### CODE REVIEW

File: app/jobs/history/ledger_importer_job.rb [hunk 1]
```diff
@@ -377,27 +374,73 @@ def import_history_effects_for_operation(sctx, scop, scresult, hop)
...
+        effects.create!("account_flags_updated", source_account, flag_changes)
+      end
+
+      if scop.master_weight.present?
+        #TODO: BLOCKED stellar-core: differentiate signer_updated and signer_added
+        #for master signer
+        effect = scop.master_weight == 0 ? "signer_removed" : "signer_updated"
+
+        effects.create!(effect, source_account, {
+          public_key: Stellar::Convert.pk_to_address(source_account),
```
Comment (masonforest) [inline app/jobs/history/ledger_importer_job.rb]: Would it make sense to add an `address` method on `Account`, `Signer` etc?

File: app/models/history/effect_factory.rb [hunk 2]
```diff
@@ -20,11 +20,15 @@ def create!(type_name, account_public_key, details)
       where(address: Stellar::Convert.pk_to_address(account_public_key)).
       first
 
+    type = History::Effect::BY_NAME[type_name]
```
Comment (masonforest) [inline app/models/history/effect_factory.rb]: Will `History::Effect::BY_NAME.fetch(type_name)` do what you want?

**Comment by masonforest** [comment]
Couple little things if you want then looks good to me :smile:

File: app/jobs/history/ledger_importer_job.rb (hunk 1 above)
Comment (nullstyle) [inline app/jobs/history/ledger_importer_job.rb]: Yeah, it definitely does.  However, that's actually a change to https://github.com/stellar/ruby-stellar-base, a dependency of horizon.

`Stellar::Convert.pk_to_address` is the current canonical helper for this functionality, so I'll keep it the same for now, but when I go about adding `address` to `Stellar::AccountID` I'll switch it over.

Thanks!

File: app/models/history/effect_factory.rb (hunk 2 above)
Comment (nullstyle) [inline app/models/history/effect_factory.rb]: Yeah, that's much better.  Thanks!  I always forget `fetch` raises an error... I tend to just use it for the default arg.

**Comment by nullstyle** [comment]
@masonforest Thanks so much for the review!  Cheers :)

**Pull request merged**